packages/core/addon/components/navi-cell-renderers/dimension.js

```diff
@@ -11,88 +11,85 @@
  */
 
 import Component from '@ember/component';
-import { computed, get } from '@ember/object';
+import { computed } from '@ember/object';
 import { readOnly } from '@ember/object/computed';
 import layout from '../../templates/components/navi-cell-renderers/dimension';
 import { isEmpty } from '@ember/utils';
+import { layout as templateLayout, tagName } from '@ember-decorators/component';
 
-export default Component.extend({
-  layout,
-
-  /**
-   * @property {Array} classNames - list of component class names
-   */
-  classNames: ['table-cell-content', 'dimension'],
-
+@templateLayout(layout)
+@tagName('')
+export default class DimensionCellRenderer extends Component {
   /**
    * @property {String} name - dimension name
    */
-  name: readOnly('column.attributes.name'),
+  @readOnly('column.attributes.name')
+  name;
 
   /**
    * @property {String} descField - field to find the description
    */
-  descField: computed('name', function() {
-    return get(this, 'name') + '|desc';
-  }),
+  @computed('name')
+  get descField() {
+    return `${this.name}|desc`;
+  }
 
   /**
    * @property {String} idField - field to find id
    */
-  idField: computed('name', function() {
-    return get(this, 'name') + '|id';
-  }),
+  @computed('name')
+  get idField() {
+    return `${this.name}|id`;
+  }
 
   /**
    * @property {Object} dimensionField - field to find dimension field
    */
-  dimensionField: computed('name', 'column.attributes.field', function() {
-    let field = get(this, 'column.attributes.field');
-    return field ? `${get(this, 'name')}|${field}` : null;
-  }),
+  @computed('name', 'column.attributes.field')
+  get dimensionField() {
+    const field = this.column?.attributes.field;
+    return field ? `${this.name}|${field}` : null;
+  }
 
   /**
    * @property {String} title - value that should be used in hoverover title
    */
-  title: computed('dimensionField', 'descField', 'idField', 'data', function() {
-    if (get(this, 'dimensionField')) {
+  @computed('dimensionField', 'descField', 'idField', 'data')
+  get title() {
+    if (this.dimensionField) {
       return '';
     }
 
-    let descField = get(this, 'descField'),
-      idField = get(this, 'idField'),
-      data = get(this, 'data');
+    const { descField, idField, data } = this;
 
-    if (!isEmpty(data[idField]) && !isEmpty(data[descField])) {
+    if (!isEmpty(data?.[idField]) && !isEmpty(data[descField])) {
```

Comment:
```suggestion
    if (!isEmpty(data?.[idField]) && !isEmpty(data?.[descField])) {
```

Reply:
I left that one out because that clause will never be evaluated if the first clause resolves to false